### PR TITLE
Remove RerouteService from ClusterService

### PR DESCRIPTION
**Pull request by thecoop**
`ClusterService` doesn't actually use `RerouteService` at all. So remove it, and change users to get `RerouteService` directly.

### CODE REVIEW

**Review by DaveCTurner — APPROVED** [review]
IIRC we did this to avoid adding yet another constructor arg to `createComponents()`, but now this is much better

File: server/src/main/java/org/elasticsearch/plugins/Plugin.java [hunk 1]
```diff
@@ -78,6 +79,11 @@ public interface PluginServices {
          */
         ClusterService clusterService();
 
+        /**
+         * A service to reroute shards to other nodes
+         */
+        RerouteService rerouteService();
+
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/plugins/Plugin.java:86]: Out of curiosity, why PluginServices is an interface, not a record?

**Review by idegtiarenko — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/plugins/Plugin.java (hunk 1 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/plugins/Plugin.java:86]: An interface allows more flexibility for `Plugin` implementors for testing, mocking etc

**Comment by thecoop** [comment]
@elasticmachine update branch

**Review by ldematte — APPROVED** [review]
LGTM too, I noticed these clusterService.getRerouteService() in a previous PR and they looked off

**Pull request merged**